pkg/controller/jobs/jobset/jobset_multikueue_adapter_test.go

```diff
@@ -0,0 +1,243 @@
+			operation: func(ctx context.Context, adapter *multikueueAdapter, managerClient, workerClient client.Client) error {
+				if isManged, _, _ := adapter.IsJobManagedByKueue(ctx, managerClient, types.NamespacedName{Name: "jobset1", Namespace: TestNamespace}, kueuealpha.MultiKueueControllerName); isManged {
+					return errors.New("expecting false")
+				}
+				return nil
+			},
+		},
+		"job with wrong managedBy is not considered managed": {
+			managersJobSets: []jobsetapi.JobSet{
+				*baseJobSetBuilder.DeepCopy().Obj(),
+			},
+			operation: func(ctx context.Context, adapter *multikueueAdapter, managerClient, workerClient client.Client) error {
+				if isManged, _, _ := adapter.IsJobManagedByKueue(ctx, managerClient, types.NamespacedName{Name: "jobset1", Namespace: TestNamespace}, kueuealpha.MultiKueueControllerName); isManged {
+					return errors.New("expecting false")
+				}
+				return nil
+			},
+			wantManagersJobSets: []jobsetapi.JobSet{
+				*baseJobSetBuilder.DeepCopy().Obj(),
+			},
+		},
+
+		"job managedBy multikueue": {
+			managersJobSets: []jobsetapi.JobSet{
+				*baseJobSetManagedByKueueBuilder.DeepCopy().Obj(),
+			},
+			operation: func(ctx context.Context, adapter *multikueueAdapter, managerClient, workerClient client.Client) error {
+				if isManged, _, _ := adapter.IsJobManagedByKueue(ctx, managerClient, types.NamespacedName{Name: "jobset1", Namespace: TestNamespace}, kueuealpha.MultiKueueControllerName); !isManged {
+					return errors.New("expecting true")
+				}
+				return nil
+			},
+			wantManagersJobSets: []jobsetapi.JobSet{
+				*baseJobSetManagedByKueueBuilder.DeepCopy().Obj(),
+			},
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			managerBuilder := utiltesting.NewClientBuilder(jobsetapi.AddToScheme).WithInterceptorFuncs(interceptor.Funcs{SubResourcePatch: utiltesting.TreatSSAAsStrategicMerge})
+			managerBuilder = managerBuilder.WithLists(&jobsetapi.JobSetList{Items: tc.managersJobSets})
+			managerBuilder = managerBuilder.WithStatusSubresource(slices.Map(tc.managersJobSets, func(w *jobsetapi.JobSet) client.Object { return w })...)
+			managerClient := managerBuilder.Build()
+
+			workerBuilder := utiltesting.NewClientBuilder(jobsetapi.AddToScheme).WithInterceptorFuncs(interceptor.Funcs{SubResourcePatch: utiltesting.TreatSSAAsStrategicMerge})
+			workerBuilder = workerBuilder.WithLists(&jobsetapi.JobSetList{Items: tc.workerJobSets})
+			workerClient := workerBuilder.Build()
+
+			ctx, _ := utiltesting.ContextWithLog(t)
+
+			adapter := &multikueueAdapter{}
+
+			gotErr := tc.operation(ctx, adapter, managerClient, workerClient)
+
+			if diff := cmp.Diff(tc.wantError, gotErr, cmpopts.EquateErrors()); diff != "" {
+				t.Errorf("unexpected error (-want/+got):\n%s", diff)
+			}
+
+			gotManagersJobSets := &jobsetapi.JobSetList{}
+			if err := managerClient.List(ctx, gotManagersJobSets); err != nil {
+				t.Errorf("unexpected list manager's jobsets error %s", err)
```

Comment:
If the List returns an error we can interrupt the test with `t.Failure`, then we don't need to use `else below.

Reply:
I prefer to continue and check the worker's list as well.

Reply:
as in the other comment